pubspec.yaml

```diff
@@ -11,16 +11,27 @@ environment:
 dependencies:
   flutter:
     sdk: flutter
+  flutter_web_plugins:
+    sdk: flutter
+  js: ^0.6.3
```

Comment:
does this need to indented?

Reply:
it seems to me that it is necessary to be indented as it is written in [How to Write a Flutter Web Plugin]( https://medium.com/flutter/how-to-write-a-flutter-web-plugin-5e26c689ea1.):

```
name: url_launcher_web
version: 0.0.1

flutter:
  plugin:
    platforms:
      web:
        pluginClass: UrlLauncherPlugin
        fileName: url_launcher_web.dart

dependencies:
  flutter:
    sdk: flutter
  flutter_web_plugins:
    sdk: flutter

environment:
  sdk: ">=2.0.0-dev.28.0 <3.0.0"
  flutter: ">=1.5.0 <2.0.0"
```

I'm still new in flutter development and I always saw dependencies indented therefore I'm sorry if  I'm misunderstanding your question. 
I also read this : 

> "Each node must be indented further than its parent node. All sibling nodes must use the exact same indentation level. However the content of each sibling node may be further indented independently. "
([YAML Specs v1.2 3rd edition](https://yaml.org/spec/1.2/spec.html#id2777534).)

Doesn't know if this can help you but js package is on pub.dev (and allows to call js API in dart code) while flutter_web_plugins is in the flutter API (and provides the Registrar class implementing for web implementations).

Reply:
I just want to make sure we are not having the js dependency in my mobile apps that do not use web.

Reply:
Hi, as one can read here : https://developer.android.com/studio/build/shrink-code
The unused dependencies are shrink at compile time when creating a release.
So don't worry to much about this JS deps. It will ultimately be removed when creating any release of a mobile application.